examples/persistable.php

```diff
@@ -0,0 +1,151 @@
+<?php
+declare(strict_types=1);
+
+namespace MongoDB\Examples;
+
+use MongoDB\BSON\ObjectId;
+use MongoDB\BSON\Persistable;
+use MongoDB\Client;
+use MongoDB\Model\BSONArray;
+use UnexpectedValueException;
+
+use function array_search;
+use function dirname;
+use function getenv;
+use function var_dump;
+
+require dirname(__FILE__) . '/../vendor/autoload.php';
+
+class PersistableEntry implements Persistable
+{
+    /** @var ObjectId */
+    private $id;
+
+    /** @var string */
+    private $name;
+
+    /** @var array<PersistableEmail> */
+    private $emails = [];
+
+    public function __construct(string $name)
+    {
+        $this->id = new ObjectId();
+        $this->name = $name;
+    }
+
+    public function getId(): ObjectId
+    {
+        return $this->id;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
+
+    public function getEmails(): array
```

Comment:
Should this utilize a doc block like the `private $emails` member for Psalm to enforce the return type as a templated array, or is that unnecessary?